### ISSUE TITLE
Simplify / clarify rescaleX and rescaleY.

### ISSUE THREAD

**Issue by altocumulus** [d3/d3-zoom#173]
This is inspired by the interesting question [*Map function in D3, confused about multiple functions passed in*](https://stackoverflow.com/questions/44135218/map-function-in-d3-confused-about-multiple-functions-passed-in/56599663?noredirect=1#comment99822564_56599663) which was posted on StackOverflow more than 2 years and ago and is still unresolved. That question asks why at some locations involving d3-zoom a scale object is passed as the `this` context to a scale's `.invert()` method like in this [Block](https://bl.ocks.org/mbostock/34f08d5e11952a80609169b7917d4172).

    x.domain(s.map(x2.invert, x2));   // 2nd argument; why set this to x2?

As `scale.invert()` does not make any use of `this` the call will work just as well if the second argument is omitted. This looked like a small glitch in just one demo at first, interestingly though, it also shows up in the d3-zoom module's own source code, namely, in the methods [`transform.rescaleX()`](https://github.com/d3/d3-zoom#transform_rescaleX) and [`transform.rescaleY()`](https://github.com/d3/d3-zoom#transform_rescaleY). Those functions' bodies are like this:

    return x.copy().domain(x.range().map(this.invertX, this).map(x.invert, x));

Right in the last call while passing `x.invert` as the callback to the `.map()` method, the `x`&mdash;the scale instance&mdash;is passed as the second argument whereby setting the `this` context. Again, this is not needed nor is it a real error. 

Long story short: Is there a deeper reason for explicitly passing in the scale as the `this` context to the `.invert()` method or can we get rid of it for the sake of simplicity and clarity?

**Comment by Fil** [comment]
The reference commit would be
https://github.com/d3/d3-zoom/commit/a116a77ebd496495e25070bee6294824a5e2feaf

**Comment by altocumulus** [comment]
@Fil Any idea on *why* that was introduced? The commit message just reads "Update README."...

**Comment by Fil** [comment]
My guess is that some scales' invert function might need to access *this*, so this would be a proper generic way of calling them.

**Comment by mbostock** [comment]
The answer is: it’s good practice in JavaScript. If the object’s methods happen to be bound to the object, it’s unnecessary to specify *this*, but one generally shouldn’t assume so unless it is documented (and not likely to change in a future version).
